Video stream fixes

bdaroz opened the pull request:
Two commits - first, correct the erroneous log output stating that the longest video stream is selected when more than one is present. Second, when there are multiple video streams, copy the ones not converted into the output. This covers cases where cover art may be stored as an mjpeg "video" stream and prevents it from being silently dropped on conversion:

```
    Stream #0:4: Video: mjpeg, yuvj444p(pc, bt470bg/unknown/unknown), 600x882, SAR 1:1 DAR 100:147, 30k fps, 1k tbr, 1k tbn, 1k tbc
    Metadata:
      FILENAME        : cover.jpg
      MIMETYPE        : image/jpeg
```

bdaroz commented:
Hold this PR. Something in commit 9f66164c0c2e93d3df18cc561f354f942add8a82 is now adding a `-filter:v scale=w='if(gt(sar,1),iw*sar,iw)':h='if(lt(sar,1),ih/sar,ih)',setsar=1` parameter which is breaking for any `copy` streams.

bdaroz commented:
Found it, line 668:

```
command.keepDAR();
```

Applies this filter to all video streams. I'll take another look at this later.

FallingSnow commented:
I added that yesterday, it forces ffmpeg to maintain the aspect ratio when scaling, another new feature I just added.

bdaroz commented:
Ok, so two problems. First the `keepDAR()` function attempts to apply the scale to all video streams. Second, and the deal breaker here, apparently ffmpeg silently breaks attachments on `copy`. Output from  `ffprobe` is substantially the same, however on further look with `mkvtools` the old attachment streams are no longer flagged as attachments. This occurs even on a simple `ffmpeg -i file.mkv -c copy -map 0 file2.mkv` command.

With that I've reverted the changes in 92961c47a7420aea892075b061d83052e3406100 and made the alert even more verbose.

I also added a global `-c:t copy` parameter to ffmpeg. In theory, should ffmpeg/ffprobe begin to detect/handle mkv attachments correctly in the future, the existing code that maps `data.streams.otherStreams` may start working.

FallingSnow commented:
This looks good. Does this fix #42?

bdaroz commented:
No, I'd call #42 an upstream bug that's not readily worked around at this level. Until ffmpeg identifies and copies attachments correctly, instead of just looking like it does, there's nothing more to do. Hopefully if ffmpeg gets fixed (I'm opening an upstream bug tomorrow after I get more logs) the changes here `-c:t copy` specifically might make it _just work_.
